Tasks/JavaToolInstaller/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -0,0 +1,145 @@
+import taskLib = require('vsts-task-lib/task');
+import tr = require('vsts-task-lib/toolrunner');
+import path = require('path');
+
+export class JavaFilesExtractor {
+    public destinationFolder: string;
+    public win: RegExpMatchArray;
+
+    // 7zip
+    public xpSevenZipLocation: string;
+    public winSevenZipLocation: string = path.join(__dirname, '7zip/7z.exe');
+
+    constructor() {
+        this.win = taskLib.osType().match(/^Win/);
+        taskLib.debug('win: ' + this.win);
+    }
+
+    private getSevenZipLocation(): string {
+        if (this.win) {
+            return this.winSevenZipLocation;
```

Comment:
Why can't you use taskLib.which to find 7z.exe on windows as well? Hard coding the path just seems a bit fragile.

Reply:
I'm not sure. I got a lot of errors when I tried to use taskLib.which().  When I did a global search I saw that all of the tasks that use 7zip have this same logic for windows environments, so I assumed that it must be a known issue.

Reply:
Okay, sounds like there is a good reason. Thanks for the explanation.